fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/leaderboard/TimeWindowScanComparisons.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * TimeWindowScanComparisons.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2022 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.provider.foundationdb.leaderboard;
+
+import com.apple.foundationdb.annotation.API;
+import com.apple.foundationdb.record.EvaluationContext;
+import com.apple.foundationdb.record.IndexScanType;
+import com.apple.foundationdb.record.provider.foundationdb.FDBRecordStoreBase;
+import com.apple.foundationdb.record.provider.foundationdb.IndexScanComparisons;
+import com.apple.foundationdb.record.query.plan.ScanComparisons;
+import com.apple.foundationdb.record.query.plan.temp.explain.Attribute;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Extend {@link IndexScanComparisons} to have {@link TimeWindowForFunction}.
+ */
+@API(API.Status.EXPERIMENTAL)
+public class TimeWindowScanComparisons extends IndexScanComparisons {
+    @Nonnull
+    private final TimeWindowForFunction timeWindow;
+
+    public TimeWindowScanComparisons(@Nonnull TimeWindowForFunction timeWindow, @Nonnull ScanComparisons comparisons) {
+        super(IndexScanType.BY_TIME_WINDOW, comparisons);
+        this.timeWindow = timeWindow;
+    }
+
+    @Nonnull
+    @Override
+    public TimeWindowScanRange bind(@Nonnull final FDBRecordStoreBase<?> store, @Nonnull final EvaluationContext context) {
+        return new TimeWindowScanRange(timeWindow.getLeaderboardType(context), timeWindow.getLeaderboardTimestamp(context), super.bind(store, context).getScanRange());
+    }
+    
+    @Override
+    public int planHash(@Nonnull final PlanHashKind hashKind) {
+        return super.planHash(hashKind) + timeWindow.planHash(hashKind);
```

Comment:
This also seems like it should be `PlanHashable.planHash` in some way instead of simple addition. I guess there's also `PlanHashable.combineHashes`, if the idea is that `PlanHashable.planHash` would cause infinite recursion.